### DISCUSSION TITLE
Переустановка общего поиска

### DISCUSSION THREAD

**Discussion by smplseven** [totumonline/totum-ru-issues-and-discussions#285] [Как это сделать?] [answered]
_Контекст:
Переезжали с одного хостинга на другой .Пришлось переустанавливать meiliserch ,но так получилось что артефакты остались  в БД поиска (это предположение)
Теперь когда ввожу запрос в поиск, то он кроме искомого значения и значения с опечаткой ,отображает то ,что не входит в область самого поиска. 
Пробовали удалить лишние значения в настройках поиска - не помогло.
В БД meiliserch - где что чистить ,тоже не совсем понятно._

Поэтому хотел бы узнать как переустановить поиск (т.к. попали артефакты в индекс и не удаляются из тотума) и не сломать текущие настройки с кнопками ( как сохранить кнопки и переиндексировать все целиком)?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
В таблице `ttm__search_settings` кнопка `Delete from index` — позволяет вручную удалить строки из индекса.

Если передать `table_id/*ALL*` то будет удален индекс поиска по всей указанной таблице.

Потом юзаем `index_it` по нужной таблице — тотум отправит все по таблице в базу поиска повторно.

Что бы понять что там за дубли можно заюзать кнопку `h_search_it` — будет такой результат:

![photo_2022-12-08 17 19 30](https://user-images.githubusercontent.com/55755565/206469864-37eb3695-5c5e-4ca9-a504-caa139474830.jpeg)

`PK` это `Primary key` — по ним все хранится в базе поиска.

Можно руками удалить по PK сделав в этой таблице кнопку:

```
=: exec(code: 'h_connect_code'; var: "posts" = $jsonPks; var: "path"= str`"/indexes/"+#h_index_name+"/documents/delete-batch"`)
    jsonPks: jsonCreate(data: json`["40-1", "40-2"]`)
```

+ `jsonPks` — `PK` на удаление

**Reply by smplseven** [reply]
Большое спасибо !
